### ISSUE TITLE
Rendering parent node label when using nested nodes

### ISSUE THREAD

**Issue by martin-ericsson** [reaviz/reaflow#68]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Current behavior

I can't seem to find a way of rendering labels for the parent node of a nested node. 

## Expected behavior

A label should be displayed for the parent node.

## What is the motivation / use case for changing the behavior?

Having a label on a parent node is quite a common use case when using graphs for documenting for instance cloud infrastructure

**Comment by amcdnl** [comment]
Hmmm - I just re-ran through the code and couldn't find anything specific that would cause this. Perhaps its hidden behind an element?

**Comment by martin-ericsson** [comment]
Yes, you are right - when I set the `elk.nodeLabels.placement` layout option to "OUTSIDE", the labels appear. 

A follow up question then: is it possible to have different layout options for different nodes? I would like the child node to still have the original "INSIDE" value?

**Comment by amcdnl** [comment]
I don't think its possible at the moment but wouldn't be hard - just need to expose the ability to set the label and then tweak the math on the layout.

**Comment by C0DK** [comment]
Basically just a way to set the position of the label, would be greatly appreciated. Labeling nested nodes are pretty difficult as is.

**Comment by C0DK** [comment]
Just a follow up..

I've done a bunch of debugging now. Damn that was a pain. So i had a difficult time understanding the syntax of https://www.eclipse.org/elk/reference/options/org-eclipse-elk-nodeLabels-placement.html

Looking at how ELJS (which you depend on) use it, i realized you have to place the three labels, and any typo will result in it moving the label to the top.
IE these are all valid
`'INSIDE V_CENTER H_LEFT'`
`'OUTSIDE V_TOP H_LEFT'`
`'INSIDE V_CENTER H_RIGHT'`

etc. I feel stupid. If anyone else feels stupid, then that is how it works.

It, however, doesn't seem to honor the horizontal argument. which sucks. - this might be due to padding or something else. I don't really know.

 I've tried to bug around with these values but no luck
 https://github.com/reaviz/reaflow/blob/d078a1fd7d0d8411bb116b95df59f56c39fed681/src/layout/elkLayout.ts#L235

One thing that did.. something, was to use settings like these on the label:
```javascript
          label={<Label
            style={{
              transform: "translateX(10px) translateY(-20px)",
              fill: 'black'
            }} />}
```

I was happy when i realized that the label class has an X and Y attribute, but those don't really do anything, and aren't honered due to elkLayout (AFAIK)

https://github.com/reaviz/reaflow/blob/1f0fffa64a767914e22e0272e6348616973d604b/src/symbols/Label/Label.tsx

I'd gladly submit a PR if you can guide me in the correct direction, but i really need labels to go to the top left corner, on "parent" nodes. and it's pretty difficult now, without some really iffy styling

**Comment by amcdnl** [comment]
Oh man - the ELK is a MAJOR pain but it is the best layout engine hands down - believe me I've tried them all.

so one note is I'd rather not rely on ELK layout for things like positioning except where I have to - reason being it would make it much more difficult to enable other engines in the future. Though I want to think the positioning engine will account for layout if it has labels and things like that so IDK if I should just give up on that over abstraction.

Let me do some quick searching and I'll give you some guidance today.

**Comment by C0DK** [comment]
I did a thing that worked:

![image](https://user-images.githubusercontent.com/9048078/110962076-d63ee300-8350-11eb-98bf-b20ad5452f24.png)

This is my canvas:
```javascript
    <Canvas
      width="100%"
      height="100%"
      fit={true}
      nodes={graph.nodes.map(newAsNode)}
      edges={graph.edges.map(asEdge)}
      layoutOptions={({
        'elk.nodeLabels.placement': 'INSIDE V_TOP H_RIGHT',
      })}
      node={
        (node) => <Node
          style={{ stroke: '#1a192b', fill: 'white', strokeWidth: 1 }}
          label={<Label
            style={{
              transform: "translateY(25px)",
              fill: 'black'
            }} />}
          onClick={() => onClickNode(node.properties.data.node)}
        />

      }
      onLayoutChange={(layout) => console.log(layout)}
    />
```

I do see your concerns, and it's a difficult architectural decision. I don't know exactly what is the right thing to do. as a "feature request" it would be nice to atleast be able to use the `'INSIDE V_TOP H_LEFT'` or maybe put an abstraction on top of that (i.e. add a "justify" and a "align" option. But i'm not sure what is making the horizontal thing fail.

What i have now is sufficient for me. Not the best, but it'll do :) i do see how much value the ELK is adding.
